src/components/comments/html-pin-adapter/index.ts

```diff
@@ -0,0 +1,884 @@
+import { Logger, Observer } from '../../../common/utils';
+import { PinMode } from '../../../web-components/comments/components/types';
+import { Annotation, PinAdapter, PinCoordinates } from '../types';
+
+import {
+  HorizontalSide,
+  TranslateAndScale,
+  Simple2DPoint,
+  SimpleParticipant,
+  TemporaryPinData,
+} from './types';
+
+export class HTMLPin implements PinAdapter {
+  // Public properties
+  // Observers
+  public onPinFixedObserver: Observer;
+
+  // Private properties
+  // Comments data
+  private annotations: Annotation[];
+  private localParticipant: SimpleParticipant = {};
+
+  // Loggers
+  private logger: Logger;
+
+  // Booleans
+  private isActive: boolean;
+  private isPinsVisible: boolean = true;
+  private movedTemporaryPin: boolean;
+
+  // Data about the current state of the application
+  private selectedPin: HTMLElement | null = null;
+  private dataAttribute: string = 'data-superviz-id';
+
+  // Coordinates/Positions
+  private mouseDownCoordinates: Simple2DPoint;
+  private commentsSide: HorizontalSide = 'left';
+  private temporaryPinCoordinates: TemporaryPinData = {};
+
+  // Elements
+  private container: HTMLElement;
+  private pinsContainer: HTMLDivElement;
+  private elementsWithDataId: Record<string, HTMLElement> = {};
+  private divWrappers: Map<string, HTMLElement> = new Map();
+  private pins: Map<string, HTMLElement>;
+  private traversedDom: Set<HTMLElement> = new Set();
+
+  // Observers
+  private resizeObserver: ResizeObserver;
+  private mutationObserver: MutationObserver;
+
+  constructor(containerId: string, dataAttributeName?: string) {
+    this.logger = new Logger('@superviz/sdk/comments-component/container-pin-adapter');
+    this.container = document.getElementById(containerId) as HTMLElement;
+
+    if (!this.container) {
+      const message = `Element with id ${containerId} not found`;
+      this.logger.log(message);
+      throw new Error(message);
+    }
+
+    this.createPinsContainer();
+    this.dataAttribute = dataAttributeName || this.dataAttribute;
+    this.isActive = false;
+    this.prepareElements();
+
+    this.mutationObserver = new MutationObserver(this.handleMutationObserverChanges);
+    this.resizeObserver = new ResizeObserver(this.handleResizeObserverChanges);
+    this.observeContainer();
+    this.observeElements();
+
+    this.pins = new Map();
+    this.onPinFixedObserver = new Observer({ logger: this.logger });
+    this.annotations = [];
+
+    document.body.addEventListener('select-annotation', this.annotationSelected);
+    Object.values(this.elementsWithDataId).forEach((element) => {
+      this.addScrollListeners(element);
+    });
```

Comment:
The elements can be modified, added or removed. You must observe these changes and add/remove the listeners. If you only add an observer in the constructor, you lose the new elements with the `dataAttribute` and don't remove the listeners of elements that are no longer in the document.

Reply:
The listeners are also added when a new element is added, inside `setElementReadyToPin`